File: maxsumofsubarray.cpp
#include<iostream>
#include<vector>
using namespace std;

int main(){
    vector<int> arr={4,-2,3,-1,6};
    int n=arr.size();
    int csum=0;
    int msum=arr[0]; 
    for(int i=0;i<n;i++){
        csum+=arr[i];
        if(csum>msum){
            msum+=csum;
        }
        else if(csum<0){
            csum=0;
        }
        cout<<"csum ->"<<csum<<" "<<"msum ->"<<msum<<endl;
    }
    cout<<"maximumsum of sub array : "<<msum;
    return 0;
}

File: maxsumofsubarray.py
arr={4,-2,3,-1,5}
n=len(arr)

csum=0
msum=0

for x in arr:
    
    csum+=x
    if csum>msum :
        msum+=csum
    elif csum<0:
        csum=0
    print(f"csum-> {csum},msum-> {msum}")

print("msum->",msum)

